### DISCUSSION TITLE
How do I preserve the field order of my schema in the database?

### DISCUSSION THREAD

**Discussion by iron3oxide** [geldata/gel#6677] [Q&A]
I am facing the following problem:
My schema includes some types that have fields that are somewhat related to each other, e.g. `first_name` and `last_name`. I therefore keep them right next to each other in the schema. In the database however (as seen in the UI data explorer), they are all over the place. 

This alone would not bug me that much as I don't plan on using the UI for anything important, but the same discombobulated order is being used when I use the python codegen feature to generate query result models I plan on using as API responses. Is there a way to fix this? Possibly related to #3792

**Comment by iron3oxide** [comment]
A somewhat working hack I found: 
1. wipe the database and delete all migrations (this will lead to strictly alphabetical ordering when you create a new migration, links will still be put at the end though). 
2. In the case of `first_name` + `last_name`: Before you create a new migration, add a property with alphabetical priority, such as `aaa` (in my case, even `complete_name` did the trick). 
3. Create a new migration, which might take a bit depending on the complexity of your schema, and apply it.
4. Since the property has alphabetical priority but requires `first_name` and `last_name`, the two will be the first fields after `id`, followed by the property. You can now delete the property from your schema and create a new migration, the order will stay.

**Reply by iron3oxide** [reply]
This only works because field order seems to depend on 

1. Field type (link or not)
2. Being required by properties with greater alphabetical priority
3. Time/order of migration that added it
4. Alphabetical priority

in roughly that order (feel free to correct me). I personally don't think this is a great solution, but maybe it has technical reasons?

**Reply by iron3oxide** [reply]
~Tried to reproduce it and it did not work again, so I guess field order is not really deterministic, which I find impractical.~

EDIT: Nvm, it does work, even with multiple properties at once, I was simply using my own recipe wrong. I was able to reorder my type this way. Let's hope I'll never have to change it while I have actual production data in the db...

**Comment by ggsammyy** [comment]
I would also like to preserve the order of fields. I'm mainly working with the UI at the moment. Have you seen anything regarding this in the past year? Thanks
